### ISSUE TITLE
Is numba a dep?

### ISSUE THREAD

**Issue by cbm755** [hanrach/p2d_fast_solver#2]
I didn't install `numba`, and can run the main codes but in `utils/banded_matrix.py` I see:

```
import numba as nb
```

**Comment by hanrach** [comment]
Right, I used banded_matrix.py before I created the custom Jacobian function. The function `diagonal_form` is not used anymore. I should probably delete this...
